src/device.rs

```diff
@@ -2426,6 +2437,38 @@ impl TaskRuntime for Runtime {
             Some(pq_event) = self.event_listeners.post_quantum_subscriber.recv() => {
                 telio_log_debug!("WG consolidation triggered by PQ event");
 
+                match (&pq_event, &self.requested_state.exit_node, &self.requested_state.last_exit_node) {
+                    (telio_pq::Event::Connecting(pubkey), Some(exit), _) if exit.public_key == *pubkey  => {
+                        let node = Node {
+                            state: PeerState::Connecting,
+                            link_state: if self.features.link_detection.is_some() {Some(LinkState::Down)} else {None},
+                            ..node_from_exit_node(exit)
+                        };
+
+                        if !self.is_dublicated_event(&node) {
+                            let _ = self.event_publishers.libtelio_event_publisher.send(
+                                Box::new(Event::Node {body: node.clone()})
+                            );
+                            self.remember_last_transmitted_node_event(node);
+                        }
+                    },
+                    (telio_pq::Event::Disconnected(pubkey), _, Some(last_exit)) if last_exit.public_key == *pubkey  => {
+                        let node = Node {
+                            state: PeerState::Disconnected,
+                            link_state: if self.features.link_detection.is_some() {Some(LinkState::Down)} else {None},
```

Comment:
```suggestion
                            link_state: link_state: self.features.link_detection.map(|_| LinkState::Down),
```

Reply:
I do not see any obvious benefit of using `.map()` in this case, especially when we are ignoring mappped type.

Reply:
It is functionally the same code just more rusty if I got it right.

Reply:
I agree with Lukas on this. At least in my opinion, the expected transformation is a conditional transformation for `Option<T>`, not just `T`, so making this a conditional transformation to `Some(_)` is what map is for. Also, `thing.map(|_| whatever)` is IMO clearer than `if thing.is_some() { Some(whatever) } else { None }`

Reply:
Though I don't see a functional issue at all with the current state of it so in the interest of merging quickly, it can be done in the follow up PR instead

Reply:
The followup PR (https://github.com/NordSecurity/libtelio/pull/1021) would remove the need for this kind of event building altogether. Hence this code is planned to be removed as well.